core/common/src/utils.rs

```diff
@@ -64,6 +76,105 @@ where
     last_error
 }
 
+/// Use Chariott to discover a service.
+///
+/// # Arguments
+/// * `chariott_uri` - Chariott's URI.
+/// * `namespace` - The service's namespace.
+/// * `name` - The service's name.
+/// * `version` - The service's version.
+/// # `communication_kind` - The service's communication kind.
+/// # `communication_reference` - The service's communication reference.
+pub async fn discover_service_using_chariott(
+    chariott_uri: &str,
+    namespace: &str,
+    name: &str,
+    version: &str,
+    communication_kind: &str,
```

Comment:
I'd consider renaming these last two parameters to include "expected"

Reply:
changed